src/events/start-blockchain-event-listeners.js

```diff
@@ -1,41 +1,83 @@
 "use strict";
 
-var async = require("async");
-var addFilter = require("./add-filter");
-var subscriptions = require("./subscriptions");
+var compact = require("lodash").compact;
 var contracts = require("../contracts");
 var ethrpc = require("../rpc-interface");
 var isFunction = require("../utils/is-function");
 var isObject = require("../utils/is-object");
 var noop = require("../utils/noop");
+var parseLogMessage = require("./parse-message/parse-log-message");
+
+function parseLogs(logs, eventByAddressAndSignature) {
+  const parsedLogs = logs.map((log) => {
+    if (eventByAddressAndSignature[log.address] === undefined) {
+      throw new Error(`Missing mapping for contract address ${log.address}`);
+    }
+    const eventDetails = eventByAddressAndSignature[log.address][log.topics[0]];
+    // Based on how we grab event logs, it's possible to get an event we don't care about if
+    // two contracts have same event, but we only want to subscribe to one of them
+    if (eventDetails === undefined) return null;
+    return parseLogMessage(eventDetails.contractName, eventDetails.eventName, log, eventDetails.abi);
+  });
+  return compact(parsedLogs);
+}
+
+// Iterates through eventCallbacks, generates eth filter and event signature mapping
+function getEventsDetails(eventsToSubscribe, activeContracts) {
+  var eventByAddressAndSignature = {};
+  const addresses = [];
+  const topics = new Set();
+  for (const contractName in eventsToSubscribe) {
+    if (Object.prototype.hasOwnProperty.call(eventsToSubscribe, contractName)) {
+      eventByAddressAndSignature[activeContracts[contractName]] = {};
+      addresses.push(activeContracts[contractName]);
+      for (let eventNameIndex = 0; eventNameIndex < eventsToSubscribe[contractName].length; eventNameIndex++) {
+        const eventName = eventsToSubscribe[contractName][eventNameIndex];
+        const eventAbi = contracts.abi.events[contractName][eventName];
+        if (!eventAbi || !eventAbi.contract || !eventAbi.signature || !eventAbi.inputs) throw new Error(`Missing abi for ${contractName}.${eventName}`);
+        eventByAddressAndSignature[activeContracts[contractName]][eventAbi.signature] = {
+          contractName: contractName,
+          eventName: eventName,
+          abi: eventAbi.inputs,
+        };
+        topics.add(eventAbi.signature);
+      }
+    }
+  }
+  const filter = {
+    address: addresses,
+    topics: [Array.from(topics)],
```

Comment:
Double checked what I'm sure you already knew: augur.js supports `Array.from` via build using babel/polyfill.